### ISSUE TITLE
Validate primaryIdentifier is unique against all resources in a template

### ISSUE THREAD

**Issue by kddejong** [aws-cloudformation/cfn-lint#3021]
### Is this feature request related to a new rule or cfn-lint capabilities?

rules

### Describe the feature you'd like to request

As we convert to the schema there is new properties.  We can look at `primaryIdentifier` (minus `readOnlyProperties`) and validate those primaryIdentifiers are unique against all resources in a template.

Example of this would be `RoleName` in `AWS::IAM::Role`

### Describe the solution you'd like

A new rule that validates the uniqueness of primaryIdentifiers across all resources of a type

### Additional context

`root already exists in stack arn:aws:cloudformation:us-east-1:0123456789012:stack/stack-name/unique-id`
for template
```yaml
Resources:
  RootRole:
    Type: 'AWS::IAM::Role'
    Properties:
      AssumeRolePolicyDocument:
        Version: "2012-10-17"
        Statement:
          - Effect: Allow
            Principal:
              Service:
                - ec2.amazonaws.com
            Action:
              - 'sts:AssumeRole'
      Path: /
      RoleName: root
      Policies:
        - PolicyName: root
          PolicyDocument:
            Version: "2012-10-17"
            Statement:
              - Effect: Allow
                Action: '*'
                Resource: '*'
  RootRole2:
    Type: 'AWS::IAM::Role'
    Properties:
      AssumeRolePolicyDocument:
        Version: "2012-10-17"
        Statement:
          - Effect: Allow
            Principal:
              Service:
                - ec2.amazonaws.com
            Action:
              - 'sts:AssumeRole'
      Path: /
      RoleName: root
      Policies:
        - PolicyName: root
          PolicyDocument:
            Version: "2012-10-17"
            Statement:
              - Effect: Allow
                Action: '*'
                Resource: '*'
```

### Is this something that you'd be interested in working on?

- [X] 👋 I may be able to implement this feature request

### Would this feature include a breaking change?

- [X] ⚠️ This feature might incur a breaking change

**Comment by PatMyron** [comment]
Duplicate of https://github.com/aws-cloudformation/cfn-lint/issues/1326

**Comment by kddejong** [comment]
Whoops closing for the other ticket.
